### PR TITLE
making the ECS init log level configurable

### PR DESCRIPTION
**Pull request by Yiyuanzzz**
### Summary

Currently, the ECS agent logging level can be configured using the ECS_LOGLEVEL option.  The ECS init does not honor this option and always logs at info level only. This pr is to address this gap and make the ECS init log level configurable as well. 

### Implementation details

- Removed scripts/bundle_log_config.sh and create logger file based on our existing logger file under ecs-init/volumes/logger
- ECS init logging level can also be configured using the ECS_LOGLEVEL env variable

- Example logs
```
level=info time=2024-02-23T03:28:11Z msg="pre-start: enabling loopback routing"
level=debug time=2024-02-23T03:28:11Z msg="testing"
level=info time=2024-02-23T03:28:11Z msg="pre-start: disabling ipv6 router advertisements"
level=debug time=2024-02-23T03:28:11Z msg="testing"
level=info time=2024-02-23T03:28:11Z msg="pre-start: creating credentials proxy route"
level=debug time=2024-02-23T03:28:11Z msg="testing"
level=info time=2024-02-23T03:28:11Z msg="pre-start: checking ecs agent container image loaded presence"
level=info time=2024-02-23T03:28:11Z msg="pre-start: ecs agent co
```

### Testing

Launched an instance with ECS_INIT_LOGLEVEL=debug and checked ecs-init.log is expected

New tests cover the changes:  yes

### Description for the changelog

Enhancement - make the ECS init log level configurable
 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -259,6 +259,7 @@ additional details on each available environment variable.
...
 | `ECS_TASK_PIDS_LIMIT` | `100` | Specifies the per-task pids limit cgroup setting for each task launched on the container instance. This setting maps to the pids.max cgroup setting at the ECS task level. See https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v2.html#pid. If unset, pids will be unlimited. Min value is 1 and max value is 4194304 (4*1024*1024) | `unset` | Not Supported on Windows |
 | `ECS_EBSTA_SUPPORTED` | `true` | Whether to use the container instance with EBS Task Attach support. ecs-init sets this variable for the ECS Agent if the instance can support mounting EBS volumes or not. ECS only schedules EBSTA tasks if this feature is supported by the platform type | `true` | `true` |
+| `ECS_INIT_LOGLEVEL`  | &lt;crit&gt; &#124; &lt;error&gt; &#124; &lt;warn&gt; &#124; &lt;info&gt; &#124; &lt;debug&gt; | The level of detail to be logged. | info | info |
```
Comment (prateekchaudhry) [inline README.md:262]: I understand this is an ecs-init config? This should probably be added to the ecs-init variables table below

File: ecs-init/logger/log.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (prateekchaudhry) [inline ecs-init/logger/log.go:1]: The year seems to vary with files, and many files do not have it. Can the year be removed?

File: ecs-init/logger/log.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package logger
+
+import (
+	"fmt"
```
Comment (prateekchaudhry) [inline ecs-init/logger/log.go:17]: nit: maybe standard library imports can be imported together as a block first followed by other imports, following go styleguide

File: README.md (hunk 1 above)
Comment (Yiyuanzzz) [reply README.md:262]: I see, will move it to table below!

File: ecs-init/logger/log.go (hunk 2 above)
Comment (Yiyuanzzz) [reply ecs-init/logger/log.go:1]: sure! will remove it

File: README.md (hunk 1 above)
Comment (Yiyuanzzz) [reply README.md:262]: good call out,  re-use ECS_LOGLEVEL would be a better idea, will do

File: ecs-init/logger/log.go (hunk 3 above)
Comment (Yiyuanzzz) [reply ecs-init/logger/log.go:17]: got it thanks!

**Comment by sparrc** [comment]
I am wondering why don't we just use the existing ECS_LOGLEVEL and apply it to the init process too? Doesn't seem particularly high-risk to add slightly more scope to that env var

**Review by prateekchaudhry — APPROVED** [review]

**Review by sparrc — APPROVED** [review]

**Pull request merged**